packages/core/src/new-api/internal/new-execution/reducers/utils.ts

```diff
@@ -0,0 +1,7 @@
+import { IgnitionError } from "../../../../errors";
+
+export function assertUnknownAction<T>(actionType: never): T {
```

Comment:
Let's try not to add this. If we design everything carefully we shouldn't need neither this nor type guards in the execution state and reducers.